### ISSUE TITLE
Beaker occasionally crashes with ~50 portals

### ISSUE THREAD

**Issue by lsjroberts** [Rotonde/beaker#28]
I've added 51 user's dats (and 44 online at this time) and occasionally Beaker crashes. I'm not sure if this is a bug in Beaker or Rotonde.

@pfrazee thoughts? Should it be able to handle this many connections? I have noticed some general slowdown in hitting other unrelated dat urls after visiting my rotonde.

**Comment by pfrazee** [comment]
Known issue, unfortunately. There's a race condition in electron's handling of custom protocols.

Beaker issue: https://github.com/beakerbrowser/beaker/issues/474
Electron issue: https://github.com/electron/electron/issues/9342

It'll be fixed by 0.8's official release.

**Comment by neauoire** [comment]
Is the trick to limit the number of profiles we follow? @lsjroberts, do you think that if all users has the latest build the crashes would stop?

**Comment by pfrazee** [comment]
Having more content basically increases the odds of hitting the race condition. It's the kind of bug which I like to call, "a total f*ing bummer"

**Comment by lsjroberts** [comment]
This PR #29 does seem to have made it less likely to crash on my own profile. It only affects the rendering not fetching any data, but I would guess it doesn't help having the dom keep splatting over itself 50 times in a few milliseconds while doing requests. But I don't really know enough about that side of things.

**Comment by pfrazee** [comment]
Yes reducing rendering would improve it. The race condition occurs when you
use the `dat://` protocol, during navigation or via embeds

On Sat, Oct 14, 2017 at 12:38 PM, Laurence Roberts <notifications@github.com

**Comment by lsjroberts** [comment]
Ah interesting. I'll bear that in mind.

**Comment by neauoire** [comment]
It's a lot more stable for me now!

**Comment by frozenpandaman** [comment]
Not sure if this is a separate issue or not, but Beaker Browser is crashing for me (Mac OS X 10.11) constantly when I'm following certain users' portals/profiles (e.g. @tha_rami's) and when navigating to my own home portal (Menu > Library > Rotonde, or manually navigating to my own dat URL). I can see the text "Fetching 5 feeds…" underneath the command box for a split second after which a crash (95% of the time) immediately follows… on rare occasions the page will load, but refreshing it once or twice will then cause a crash. Removing the dat:// entry from the list associated with the`port` key in `portal.json` seems to fix it. I'm still able to navigate to these portals manually or via hyperlink – the crash only seems to occur when certain portals are listed in the sidebar of my home portal/dat site.

**Comment by lsjroberts** [comment]
@frozenpandaman Definitely sounds related, I'll take a look and see if I can spot some points to optimise. Sounds like there may not be too much we can do until Beaker 0.8 is out though.
